packages/main/src/plugin/kubernetes-context-state.ts

```diff
@@ -0,0 +1,243 @@
+    private readonly apiSender: ApiSenderType,
+  ) {}
+
+  async update(enabled: boolean | undefined, kubeconfig: KubeConfig) {
+    this.kubeConfig = kubeconfig;
+    if (enabled) {
+      // Add informers for new contexts
+      for (const context of this.kubeConfig.contexts) {
+        if (this.contextsInternalState.get(context.name) === undefined) {
+          this.contextsState.set(context.name, {
+            reachable: false,
+            podsCount: 0,
+            deploymentsCount: 0,
+            replicasetsCount: 0,
+          });
+          const informers = this.createKubeContextInformers(context);
+          if (informers) {
+            this.contextsInternalState.set(context.name, informers);
+          }
+        }
+      }
+      // Delete informers for removed contexts
+      for (const [name, state] of this.contextsInternalState) {
+        if (!this.kubeConfig.contexts.find(c => c.name === name)) {
+          await state.podInformer?.stop();
+          await state.deploymentInformer?.stop();
+          await state.replicasetInformer?.stop();
+          this.contextsInternalState.delete(name);
+          this.contextsState.delete(name);
+        }
+      }
+    } else {
+      for (const state of this.contextsInternalState.values()) {
+        await state.podInformer?.stop();
+        await state.deploymentInformer?.stop();
+        await state.replicasetInformer?.stop();
+      }
+      this.contextsInternalState.clear();
+      this.contextsState.clear();
+      this.dispatchContextsState();
+    }
+    console.log('==> # informers', this.contextsInternalState.size, this.contextsState.size);
+  }
+
+  private createKubeContextInformers(context: KubeContext): ContextInternalState | undefined {
+    const kc = new KubeConfig();
+    const cluster = this.kubeConfig.clusters.find(c => c.name === context.cluster);
+    const user = this.kubeConfig.users.find(u => u.name === context.user);
+    if (!cluster || !user) {
+      return;
+    }
+    kc.loadFromOptions({
+      clusters: [cluster],
+      users: [user],
+      contexts: [context],
+      currentContext: context.name,
+    });
+
+    const ns = context.namespace ?? 'default';
+    return {
+      podInformer: this.createPodInformer(kc, ns, context),
+      deploymentInformer: this.createDeploymentInformer(kc, ns, context),
+      replicasetInformer: this.createReplicasetInformer(kc, ns, context),
+    };
+  }
+
+  private createPodInformer(kc: KubeConfig, ns: string, context: KubeContext): Informer<V1Pod> & ObjectCache<V1Pod> {
+    const k8sApi = kc.makeApiClient(CoreV1Api);
+    const listFn = () => k8sApi.listNamespacedPod(ns);
+    const informer = makeInformer(kc, `/api/v1/namespaces/${ns}/pods`, listFn);
+
+    informer.on('add', (_obj: V1Pod) => {
+      const previous = this.contextsState.get(context.name);
+      if (previous) {
+        previous.podsCount++;
+      }
+      this.dispatchContextsState();
+    });
+
+    informer.on('delete', (_obj: V1Pod) => {
```

Comment:
```suggestion
    informer.on('delete', () => {
```
My understanding the 'underscore' would be good if you have first parameter that is not used and then some more that is used. If it is the only parameter, you can just skip it.